### PR TITLE
Fix Bug 1144134: Indicator improvements

### PR DESCRIPTION
**Pull request by stephaniehobson**
Attempted to standardized colours, padding, borders, ect. on overhead indicators.

Improvements:
- combined many disparate classes to share same code
- standardized colour use
- standardized display (borders, gradients, etc)
- added icons
- made pre not italicized
- made variable for "tiny" text, standardized font
- moved theme mixins into shared mixin file for reuse

Testing pages:
- https://developer.mozilla.org/en-US/docs/User%3Astephaniehobson%3Aindicators
- https://developer.mozilla.org/en-US/docs/User%3Ateoli/In_content

### CODE REVIEW

**Comment by stephaniehobson** [comment]
Before and after view: https://bugzilla.mozilla.org/show_bug.cgi?id=1144134

Question for reviewer: does this need a waffle flag? I'm leaning towards no but I'm worried that I just think that because a waffle is more work :P

**Comment by openjck** [comment]
Travis failure looks like an intermittent thing. Restarting...

File: media/stylus/components/wiki/customcss.styl [hunk 1]
```diff
@@ -791,144 +613,14 @@ ul.card-grid li {
...
-    white-space: nowrap;
-    vertical-align: middle;
-    font-size: 12px;
-    float: right;
-}
-[dir="rtl"] .indicatorInHeadline,
-[dir="rtl"] .standardNoteMethod {
-    float: left;
-}
+/* */
```
Comment (openjck) [inline media/stylus/components/wiki/customcss.styl]: Just a nitpick. Was there a comment here that was accidentally lost?

File: media/stylus/includes/mixins.styl [hunk 2]
```diff
@@ -447,3 +447,85 @@ $code-block {
...
+/* functions to get correct text colors for variations of theme colors
+-------------------------------------------------------------- */
+
+theme-pale-text($color) {
+    $color = theme-pale-light($color);
+    $color = darken($color, 60%);
+    return $color;
+}
+
+theme-pale-contrast-text($color){
```
Comment (openjck) [inline media/stylus/includes/mixins.styl]: Very cool. :hamster:

File: media/stylus/includes/mixins_indicators.styl [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+-------------------------------------------------------------- */
+
+$indicator-inline {
+    display: inline-block;
+    min-width: 20px;
+    vertical-align: baseline;
+    bidi-style(margin-left, ($grid-spacing / 2 ), margin-right, 0);
+    border-radius: 2px;
+    bidi-style(border-left-width, 4px, border-right-width, 0);
+    bidi-style(border-left-style, solid, border-right-style, none);
```
Comment (openjck) [inline media/stylus/includes/mixins_indicators.styl]: Would the be able to be condensed into `bidi-style(border-left, ...)`?

File: media/stylus/includes/mixins_indicators.styl [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    box-theme($warning);
+}
+
+$indicator-info {
+    box-theme($default);
+}
+
+$indicator-obsolete {
+    box-theme($default);
+}
```
Comment (openjck) [inline media/stylus/includes/mixins_indicators.styl]: We could consider separating the `$indicator-info` and `$indicator-obsolete` selectors by a comma, but then again the readability of this is kinda nice, and Stylus is going to compile this anyway. Whatever you think is best. :smile:

File: media/stylus/components/wiki/customcss.styl (hunk 1 above)
Comment (stephaniehobson) [inline media/stylus/components/wiki/customcss.styl]: There was a comment and I moved it into a different file but there still remained a logical separation between the code above what I removed and the code below it.

I would have used 
```/* -------------------------------------------------------------- */```
if I was in a file that had been re-orged but I wasn't so I chameleon coded instead.

Not sure there's a right answer here, happy to change it if you have a suggestion :)

Comment (openjck) [inline media/stylus/components/wiki/customcss.styl]: Gotcha. An empty comment makes sense to me, then. :smile:

File: media/stylus/includes/mixins_indicators.styl (hunk 3 above)
Comment (stephaniehobson) [inline media/stylus/includes/mixins_indicators.styl]: I had to read the spec to see if colour was optional in the short hand declaration but, yes, yes I can. 💫

**Comment by stephaniehobson** [comment]
Updated!

**Comment by openjck** [comment]
This is fantastic! :100:

**Pull request merged**